File: app/src/main/java/ru/profapp/RanobeReader/Common/ThemeUtils.java
package ru.profapp.RanobeReader.Common;

import android.app.Activity;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;

/**
 * Created by Ruslan on 15.03.2018.
 */

public class ThemeUtils {
    private static int sTheme=AppCompatDelegate.MODE_NIGHT_NO;

    public static void change(AppCompatActivity activity) {
        activity.recreate();
    }

    public static void change(Activity activity) {
        activity.recreate();
    }

    public static void onActivityCreateSetTheme() {
        AppCompatDelegate.setDefaultNightMode(sTheme);
    }

    public static void setTheme(boolean theme) {
        sTheme = (theme) ? AppCompatDelegate.MODE_NIGHT_YES : AppCompatDelegate.MODE_NIGHT_NO;
    }

}

File: app/src/main/java/ru/profapp/RanobeReader/Common/ThemeUtils.kt
package ru.profapp.RanobeReader.Common

import android.app.Activity
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate

/**
 * Created by Ruslan on 15.03.2018.
 */

object ThemeUtils {

     private var sTheme = AppCompatDelegate.MODE_NIGHT_NO

    fun change(activity: AppCompatActivity) {
        activity.recreate()
    }

    fun change(activity: Activity) {
        activity.recreate()
    }

    fun onActivityCreateSetTheme() {
        AppCompatDelegate.setDefaultNightMode(sTheme)
    }

    fun setTheme(theme: Boolean) {
        sTheme = if (theme) AppCompatDelegate.MODE_NIGHT_YES else AppCompatDelegate.MODE_NIGHT_NO
    }

}
